test/integration/controller_instance_test.go

```diff
@@ -0,0 +1,138 @@
+
+	controller.AddGetNamespaceReaction(fakeKubeClient)
+
+	// create the secret that the Broker resource will use for auth to the fake broker server
+	fakeKubeClient.AddReactor("get", "secrets", func(clientgotesting.Action) (bool, runtime.Object, error) {
+		return true, &v1.Secret{
+			Data: map[string][]byte{
+				"username": []byte(brokerUsername),
+				"password": []byte(brokerPassword),
+			},
+		}, nil
+	})
+
+	// build the chain from Instance -> ServiceClass -> Broker
+	testBroker := faketypes.GetBroker()
+	testBroker.Spec.URL = controllerItems.BrokerServer.URL
+	testBroker.Spec.AuthInfo = &v1alpha1.BrokerAuthInfo{
+		BasicAuthSecret: &v1.ObjectReference{
+			Namespace: "test",
+			Name:      "test",
+		},
+	}
+	testServiceClass := faketypes.GetServiceClass()
+	testInstance := faketypes.GetInstance()
+
+	sharedInformers.Brokers().Informer().GetStore().Add(testBroker)
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(testServiceClass)
+
+	fakeCatalogClient.AddReactor("get", "serviceclasses", func(clientgotesting.Action) (bool, runtime.Object, error) {
+		return true, testServiceClass, nil
+	})
+	fakeCatalogClient.AddReactor("get", "brokers", func(clientgotesting.Action) (bool, runtime.Object, error) {
+		return true, testBroker, nil
+	})
+
+	// Make the provision return an error that is "unexpected"
+	fakeBrokerServerHandler.ProvisionRespError = errors.New("test provision error")
+
+	// there should be nothing that is polling since no instances have been reconciled yet
+	if testController.PollingQueueLen() != 0 {
+		t.Fatalf("Expected the polling queue to be empty")
+	}
+
+	reconcileErr := testController.ReconcileInstance(testInstance)
+	if reconcileErr == nil {
+		t.Fatalf("expected an error from reconcileInstance")
+	}
+	statusCodeErr, ok := reconcileErr.(osb.HTTPStatusCodeError)
+	if !ok {
+		t.Fatalf("expected an OSB client HTTPStatusCodeError, got %#v", reconcileErr)
+	}
+	if statusCodeErr.StatusCode != http.StatusInternalServerError {
+		t.Fatalf("expected an internal server error, got %d", statusCodeErr.StatusCode)
+	}
+
+	actions := fakeCatalogClient.Actions()
+	controller.AssertNumberOfActions(t, actions, 1)
+
+	// verify that 2 kubernetes actions occurred - a GET to the ServiceClass, then a GET to the
+	// Broker
+	kubeActions := fakeKubeClient.Actions()
+	controller.AssertNumberOfActions(t, kubeActions, 2)
+
+	updatedInstance := controller.AssertUpdateStatus(t, actions[0], testInstance)
+	controller.AssertInstanceReadyFalse(t, updatedInstance)
+
+	// there should be 1 request to the provision endpoint
+	numProvReqs := len(fakeBrokerServerHandler.ProvisionRequests)
+	if numProvReqs != 1 {
+		t.Fatalf("%d provision requests were made, expected 1", numProvReqs)
+	}
+
+	// The item should not have been added to the polling queue for later processing
+	if testController.PollingQueueLen() != 0 {
+		t.Fatalf("Expected polling queue to be empty")
+	}
+	controller.AssertAsyncOpInProgressFalse(t, updatedInstance)
```

Comment:
Let's run the controller similar to the other integration tests so that are testing the behavior of a running controller instead of exercising a specific (and very clean) ordering of methods in a test harness